index.bs

```diff
@@ -6152,37 +6152,40 @@ leakage due to such an attack:
     - When verifying an {{AuthenticatorAssertionResponse}} response from the [=authenticator=], make it indistinguishable whether
           verification failed because the signature is invalid or because no such user or credential is registered.
 
-    - Perform a different authentication step, such as username and password authentication,
-        before initiating the WebAuthn [=authentication ceremony=].
-        This moves the username enumeration problem from the WebAuthn [=authentication ceremony=]
+    - Perform a multi-step [=authentication ceremony=], e.g., beginning with supplying username and password or a session cookie,
+        before initiating the WebAuthn [=ceremony=] as a subsequent step.
+        This moves the username enumation problem from the WebAuthn step
         to the preceding authentication step, where it may be easier to solve.
 
 
 ## Privacy leak via credential IDs ## {#sctn-credential-id-privacy-leak}
 
 [INFORMATIVE]
 
-This privacy consideration applies to [=[RPS]=] supporting [=single-factor=] authentication with [=non-resident credentials=].
+This privacy consideration applies to [=[RPS]=] that support [=authentication ceremonies=]
+where a [=non-resident credential=] is used as the first authentication step.
 
 In this case the {{PublicKeyCredentialRequestOptions/allowCredentials}} argument risks leaking [PII],
-if the user can initiate an [=authentication ceremony=] by only providing a username.
+since it exposes the user's [=credential IDs=] to an unauthenticated caller.
 [=Credential IDs=] are designed to not be correlatable between [=[RPS]=],
 but the length of a [=credential ID=] might be a hint as to what type of [=authenticator=] created it.
 It is likely that a user will use the same username and set of [=authenticators=] for several [=[RPS]=],
 so the number of [=credential IDs=] in {{PublicKeyCredentialRequestOptions/allowCredentials}} and their lengths
 might serve as a global correlation handle to de-anonymize the user.
+Knowing a user's [=credential IDs=] also makes it possible to confirm guesses about the user's identity
+given only momentary physical access to one of the user's [=authenticators=].
 
-The simplest ways to prevent this information leak
-is to not support [=single-factor=] authentication with [=non-resident credentials=],
-for example by:
+The simplest way to prevent this information leak
```

Comment:
No.. With non-resident credentials also you should be able to do single-factor/2FA authentication.. 

How about we collapse this to, for authentication ceremory, add random credentials with random length to the list??

Reply:
Actually, we should talk in terms of second factor vs first factor.. and not in terms of whether credential is resident or not..

Reply:
>No.. With non-resident credentials also you should be able to do single-factor/2FA authentication..
>How about we collapse this to, for authentication ceremory, add random credentials with random length to the list??

The last paragraph of the section is about this, is it enough?

>Actually, we should talk in terms of second factor vs first factor.. and not in terms of whether credential is resident or not..

I don't think so - these are privacy considerations about exposing an `allowCredentials` list to an unauthenticated user, so it's precisely about non-resident keys whether they are MFA (UV) credentials or not.

Reply:
> Actually, we should talk in terms of second factor vs first factor.. and not in terms of whether credential is resident or not..

I agree with this idea.

The issue is about whether or not you need to send credentialIDs with allowCredentials or not. 
By sending credentialIDs, (thereby exposing credentialIDs), regardless of resident credentials or non-resident credentials, we can have the "with usename" flows that have a FIDO credential and typically User Verification-- 2 factors.

Reply:
Ok, on second thought you are both right that this would also apply to resident credentials used in second-factor mode. I'll see if I can work that in...

Reply:
I made an attempt at this in e7e11c4, please have another look and let me know what you think!